src/utils/MzMLSplitter.cpp

```diff
@@ -155,76 +155,96 @@ class TOPPMzMLSplitter :
     writeLog_("Total spectra: " + String(spectra.size()));
     writeLog_("Total chromatograms: " + String(chromatograms.size()));
 
-    Size spec_start = 0, chrom_start = 0;
-    Size width = String(parts).size();
-    for (Size counter = 1; counter <= parts; ++counter)
+    //calculate split ranges e.g., (0,100), (101, 200), ...
+    vector<std::pair<int, int>> parts_spec_ranges;
+    vector<std::pair<int, int>> parts_chrom_ranges;
+    Size spec_start = 0,  chrom_start = 0;
+    Size width = String(parts).size(); 
+    Size n_chrom, n_spec;   
+
+    for (Size counter = 0 ; counter < parts; ++counter)
     {
-      ostringstream out_name;
-      out_name << out << "_part" << setw(width) << setfill('0') << counter
-               << "of" << parts << ".mzML";
-      PeakMap part = experiment;
-      addDataProcessing_(part, getProcessingInfo_(DataProcessing::FILTERING));
-
-      Size remaining = parts - counter + 1;
-      Size n_spec = ceil((spectra.size() - spec_start) / double(remaining));
-      if (n_spec > 0)
+      Size remaining = parts - counter ;
+      //n_spec has to be substracted by -1, since list of spectra (index) starts at 0  
+      n_spec = ceil((spectra.size() - spec_start) / double(remaining));
+      n_chrom = ceil((chromatograms.size() - chrom_start) / double(remaining));
+
+      //chromatograms
+      if (n_chrom > 0)
       {
-        int last_spectrum = spec_start + n_spec;   
-        part.reserveSpaceSpectra(n_spec);  
-        for (Size i = spec_start; i < spec_start + n_spec; ++i)
+        parts_chrom_ranges.push_back(std::make_pair(chrom_start, chrom_start + n_chrom - 1));   
+      }
+
+      //spectra
```

Comment:
I know it is pedantic but space after // is easier to read. Also would be nicer to be consistent in the use of lower / upper case for first letter in comments (e.g. Last spectrum vs. chromatograms)

Reply:
It is perfectly fine to point that out - thanks.